lib/models/TypeEffective.ts

```diff
@@ -0,0 +1,116 @@
+				Bug: 1.4, Dark: 1, Dragon: 0.714, Electric: 1, Fairy: 1, Fighting: 1,
+				Fire: 0.714, Flying: 1, Ghost: 1, Grass: 1.4, Ground: 1, Ice: 1.4,
+				Normal: 1, Poison: 1, Psychic: 1, Rock: 0.714, Steel: 1.4, Water: 0.714
+			},
+			Flying: {
+				Bug: 1.4, Dark: 1, Dragon: 1, Electric: 0.714, Fairy: 1, Fighting: 1.4,
+				Fire: 1, Flying: 1, Ghost: 1, Grass: 1.4, Ground: 1, Ice: 1,
+				Normal: 1, Poison: 1, Psychic: 1, Rock: 0.714, Steel: 0.714, Water: 1
+			},
+			Ghost: {
+				Bug: 1, Dark: 0.714, Dragon: 1, Electric: 1, Fairy: 1, Fighting: 1,
+				Fire: 1, Flying: 1, Ghost: 1.4, Grass: 1, Ground: 1, Ice: 1,
+				Normal: 0.51, Poison: 1, Psychic: 1.4, Rock: 1, Steel: 1, Water: 1
+			},
+			Grass: {
+				Bug: 0.714, Dark: 1, Dragon: 0.714, Electric: 1, Fairy: 1, Fighting: 1,
+				Fire: 0.714, Flying: 0.714, Ghost: 1, Grass: 0.714, Ground: 1.4, Ice: 1,
+				Normal: 1, Poison: 0.714, Psychic: 1, Rock: 1, Steel: 1, Water: 1
+			},
+			Ground: {
+				Bug: 0.714, Dark: 1, Dragon: 1, Electric: 1.4, Fairy: 1, Fighting: 1,
+				Fire: 1.4, Flying: 0.51, Ghost: 1, Grass: 0.714, Ground: 1, Ice: 1,
+				Normal: 1, Poison: 1.4, Psychic: 1, Rock: 1.4, Steel: 1.4, Water: 1
+			},
+			Ice: {
+				Bug: 1, Dark: 1, Dragon: 1.4, Electric: 1, Fairy: 1, Fighting: 1,
+				Fire: 0.714, Flying: 1.4, Ghost: 1, Grass: 1.4, Ground: 1.4, Ice: 0.714,
+				Normal: 1, Poison: 1, Psychic: 1, Rock: 1, Steel: 0.714, Water: 0.714
+			},
+			Normal: {
+				Bug: 1, Dark: 1, Dragon: 1, Electric: 1, Fairy: 1, Fighting: 1,
+				Fire: 1, Flying: 1, Ghost: 0.51, Grass: 1, Ground: 1, Ice: 1,
+				Normal: 1, Poison: 1, Psychic: 1, Rock: 0.714, Steel: 0.714, Water: 1
+			},
+			Poison: {
+				Bug: 1, Dark: 1, Dragon: 1, Electric: 1, Fairy: 1.4, Fighting: 1,
+				Fire: 1, Flying: 1, Ghost: 0.714, Grass: 1.4, Ground: 0.714, Ice: 1,
+				Normal: 1, Poison: 0.714, Psychic: 1, Rock: 0.714, Steel: 0.51, Water: 1
+			},
+			Psychic: {
+				Bug: 1, Dark: 0.51, Dragon: 1, Electric: 1, Fairy: 1, Fighting: 1.4,
+				Fire: 1, Flying: 1, Ghost: 1, Grass: 1, Ground: 1, Ice: 1,
+				Normal: 1, Poison: 1.4, Psychic: 0.714, Rock: 1, Steel: 0.714, Water: 1
+			},
+			Rock: {
+				Bug: 1.4, Dark: 1, Dragon: 1, Electric: 1, Fairy: 1, Fighting: 0.714,
+				Fire: 1.4, Flying: 1.4, Ghost: 1, Grass: 1, Ground: 0.714, Ice: 1.4,
+				Normal: 1, Poison: 1, Psychic: 1, Rock: 1, Steel: 0.714, Water: 1
+			},
+			Steel: {
+				Bug: 1, Dark: 1, Dragon: 1, Electric: 0.714, Fairy: 1.4, Fighting: 1,
+				Fire: 0.714, Flying: 1, Ghost: 1, Grass: 1, Ground: 1, Ice: 1.4,
+				Normal: 1, Poison: 1, Psychic: 1, Rock: 1.4, Steel: 0.714, Water: 0.714
+			},
+			Water: {
+				Bug: 1, Dark: 1, Dragon: 0.714, Electric: 1, Fairy: 1, Fighting: 1,
+				Fire: 1.4, Flying: 1, Ghost: 1, Grass: 0.714, Ground: 1.4, Ice: 1,
+				Normal: 1, Poison: 1, Psychic: 1, Rock: 1.4, Steel: 1, Water: 0.714
+			}
+		}
+	}
+}
+
+export class TypeEffective extends TypeEffectiveTable {
+	look(AttackType: PokemonType, DeffendType: PokemonType) {
```

Comment:
Better naming for look function would be nice. 
`TypeEffective` class will be used something like this 
```ts
const typeEffective = new TypeEffective();
const te_multiplier = typeEffective.multiplier(attackType, DefendeType);
```


```suggestion
	multiplier(AttackType: PokemonType, DeffendType: PokemonType) {
```